### ISSUE TITLE
Use a function with nnn#pick

### ISSUE THREAD

**Issue by johngodlee** [mcchrish/nnn.vim#91]
With `nnn#action` it's possible to pass the array of results to a custom function, like so:

```
function! s:copy_results(lines)
  let joined_lines = join(a:lines, "\n")
  if len(a:lines) > 1
    let joined_lines .= "\n"
  endif
  let @+ = joined_lines
endfunction

let g:nnn#action = { '<c-y>': function('s:copy_results') }
```

Is it possible to also use a custom function with `nnn#pick`. I would like to make a custom mapping. This is what I have tried so far:

```
nnoremap <Leader>A :call nnn#pick('/Users/xxx', { 'edit': function('s:copy_results')})<CR>
```

But this produces the following error:

```
E81: Using <SID> not in a script context
E475: Invalid argument: s:copy_results
```

If I press Enter through this error nnn.vim opens and I can select files, but then when I press Enter again I get this error:

```
Error detected while processing function <SNR>89_callback[6]..<SNR>89_eval_temp_file:
line   16:
E486: Pattern not found: Users
```

TL;DR - Can I pass a custom function to `nnn#pick`

**Comment by mcchrish** [comment]
Yes, it's possible. I have a working example in my dotfile. https://github.com/mcchrish/dotfiles/blob/master/vim/Library/Preferences/nvim/after/plugin/keys.vim#L234

**Comment by johngodlee** [comment]
@mcchrish Excellent thank you for the help. In case you are interested what I wanted to achieve:

```
" Define function to attach files in Mutt
function! s:mutt_attach(lines)
  let prettylines = ''
  for i in a:lines
    let prettylines .= 'Attach: ' . fnameescape(i) . "\n"
  endfor
  6put =prettylines
endfunction

function! s:nnncall(...)
  let l:dir = get(a:, 1, '')
  let l:opts = get(a:, 2, { 'edit': 'edit' })
  let l:keypress = get(a:, 3, '')
  call nnn#pick(l:dir, l:opts)
  if strlen(l:keypress) > 0
    call feedkeys(l:keypress)
  endif
endfunction

autocmd Filetype mail nnoremap <silent> <Leader>A :call <SID>nnncall('/Users/johngodlee', { 'edit': function('<SID>mutt_attach') })<CR>
```

This allows me to attach files while editing an email for Mutt using vim. Basically it just lets me select a file(s) with nnn.vim, format the results by adding "Attach: " and sanitising the filenames, and then insert those lines at the correct point in the header material.

What I've written above works great, but if you have the inclination, any pointers to improve the vimscript would be appreciated.
